algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,107 @@
+package com.twitter.algebird
+
+object Correlation {
+  def apply(x: (Double, Double)): Correlation =
+    Correlation(c2 = 0, m2Left = 0, m2Right = 0, m1Left = x._1, m1Right = x._2, 1L)
+
+  implicit val group: Group[Correlation] = CorrelationGroup
+}
+
+/**
+ * A class to calculate covariance and the first two central moments of a sequence of pairs of Doubles, from which the
+ * pearson correlation coeifficient can be calculated.
+ *
+ * m{i}Left denotes the ith central moment of the first projection of the pair.
+ * m{i}Right denotes the ith central moment of the second projection of the pair.
+ * c2 the covariance equivalent of the second central moment, i.e. c2 = Sum_(l,r) (l - m1Left)*(r - m1Right).
+ *
+ */
+case class Correlation(
+    c2: Double,
```

Comment:
Unlike `Moments`, I listed the arguments here in descending order of degree. My reasoning was that `c2` is the most important quantity in this calculation, and should therefore be listed first, but I'm open to other opinions.